src/client/hooks/useProducts.js

```diff
@@ -1,23 +1,44 @@
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 
-const fetchProducts = () => {
-  const url = '/api/products';
+const fetchProducts = (limit, offset, sortKey, sortOrder) => {
+  const url = `/api/products?limit=${limit}&offset=${offset}&sortKey=${sortKey}&sortOrder=${sortOrder}`;
   return fetch(url).then((response) => response.json());
 };
 
-export const useProducts = () => {
-  const [products, setProducts] = useState();
-  const [isLoading, setLoading] = useState(true);
+export const useProducts = (
+  { sortKey, sortOrder } = { sortKey: 'name', sortOrder: 'asc' },
+) => {
+  const PRODUCT_PER_PAGE = 12;
+  const [products, setProducts] = useState([]);
+  const [totalCount, setTotalCount] = useState();
+  const [isLoading, setLoading] = useState(false);
   const [error, setError] = useState();
 
-  if (!products && !error) {
-    fetchProducts()
-      .then((productsResponse) => setProducts(productsResponse))
+  const loadMoreProducts = () => {
+    if (isLoading) {
+      return;
+    }
+    setLoading(true);
+
+    fetchProducts(PRODUCT_PER_PAGE, products.length, sortKey, sortOrder)
+      .then(({ items, totalCount: _totalCount }) => {
+        setTotalCount(_totalCount);
+        setProducts([...products, ...items]);
+      })
       .catch((err) => setError(err))
       .finally(() => setLoading(false));
-  }
+  };
+
+  useEffect(() => {
+    console.log(sortKey, sortOrder);
+    setProducts([]);
+    setLoading(false);
+    loadMoreProducts();
```

Comment:
I'm not sure if I understand how this is supposed to work.

Anytime sortKey or sortOrder is changed, this effect will run. Which will clear the products list.
Then you try to use the length of that list to decide which page to load next?
I don't understand if useProducts should return the current page, or all items that have been loaded so far?
It also does not seem to handle the case where `products.length === totalCount` and there is nothing more to fetch.

Further, it should not be needed to set loading to false, your initial value is false and the `finally` clause should make sure it is set to false.